projects/media-viewer/src/lib/store/reducers/bookmarks.reducer.ts

```diff
@@ -54,6 +54,38 @@ export function bookmarksReducer (state = initialBookmarksState,
         loaded: true
       }
     }
+
+    case fromBookmarks.DELETE_BOOKMARK_SUCCESS: {
+      const id: String = action.payload;
+      const bookmarks = state.bookmarks.filter(bookmark => bookmark.id !== id);
+      return {
+        ...state,
+        bookmarks,
+        loading: false,
+        loaded: true
+      }
+    }
+
+    case fromBookmarks.UPDATE_BOOKMARK_SUCCESS: {
+      const bookmark: Bookmark = action.payload;
+      const name = bookmark.name;
+      const bookmarks = state.bookmarks.map(bmark => {
```

Comment:
i am very surprised that this is not giving u an error as you are mutating state here... that means that our store-feeze is not working dam - I'll make a note of that and try to fix it. 
always decouple state by [...state.bookmarks].map...

Reply:
I don't think I am mutating the state here.. I'm just assigning the bookname.name value to a variable - but maybe this is not what you are talking about? Thank you :)